### ISSUE TITLE
ControlNet

### ISSUE THREAD

**Issue by neverix** [ashawkey/stable-dreamfusion#152]
The pose of generated characters can't currently be controlled in a fine-grained way. Adding this would allow for 3D animation to be generated as well.

**Describe the solution you'd like**
Implement [ControlNet](https://github.com/lllyasviel/ControlNet) - specifically, the pose and segmentation models.

**Describe alternatives you've considered**
Do some sort of image-to-image (or model-to-model) thing

**Comment by ashawkey** [comment]
@neverix Yes, this is interesting, I guess the pose model can be used to synthesize better human by initializing a skeleton and render it from different views as a prior. However, it is quite task-specific and I may try it later in other projects.

**Comment by ForceConstant** [comment]
@ashawkey does dreamfusion have way to use img2img instead of txt2img? Whereas you could manually create a predfined image with controlnet which turns objects all the directions, and then feed that to dreamfusion.

**Comment by neverix** [comment]
> @neverix Yes, this is interesting, I guess the pose model can be used to synthesize better human by initializing a skeleton and rendering it from different views as a prior. However, it is quite task-specific and I may try it later in other projects.

The basic approach of rendering a colored mesh would also be useful for rendering from segmentation or applying img2img. If I implemented it I would add a hook users can add that functionality into

**Comment by ForceConstant** [comment]
@ashawkey I just wanted to add how big a deal something like this would be, being able to generate real 3d characters and models using Stable Diffusion would be huge.

**Comment by neverix** [comment]
It's already possible, just not using ControlNet

**Comment by flobotics** [comment]
the biggest problem with generating good results is that stable-diffusion does not output the same object with different prompts.

If i get it right, there are words like "front view" ,"side view", etc. added to the text-prompt to generate images from other views. But if you run e.g. python nerf/sd.py "armchair front view" and "armchair side view", you see that you get two totally different armchair pictures. Also you can see, that most pictures with the tag "side view" does not show the side-view of the object, a front-view mostly instead.

i could be wrong

**Comment by neverix** [comment]
> the biggest problem with generating good results is that stable-diffusion does not output the same object with different prompts.
> 
> If i get it right, there are words like "front view" ,"side view", etc. added to the text-prompt to generate images from other views. But if you run e.g. python nerf/sd.py --seed 100 "armchair front view" and "armchair side view", you see that you get two totally different armchair pictures. Also you can see, that most pictures with the tag "side view" does not show the side-view of the object, a front-view mostly instead.
> 
> i could be wrong

Yes, ControlNet grounding could potentially help solve this. But the signal is already quite good and there are other ways to make progress on this

**Comment by Preternature** [comment]
> It's already possible, just not using ControlNet

How is it already possible?

**Comment by neverix** [comment]
> > It's already possible, just not using ControlNet
> 
> How is it already possible?

Well, you are looking at the repo that makes it possible

**Comment by sjx-ali** [comment]
> 

Hi, I meet the same problem, do you solve it?

**Comment by sjx-ali** [comment]
> I tried controlnet but get this: ![df_ep0016_0002_rgb](https://user-images.githubusercontent.com/44969890/222178477-4fab85fc-410e-44d1-b950-31914d22a09d.png) Just like use vanilla nerf, it is likely to be misconvergence

Hi, I meet the same problem, do you solve it?
